src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+
+import { GenericVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { enterVenue } from "utils/url";
+
+import { usePosters } from "hooks/posters";
+
+import { PosterPreview } from "./components/PosterPreview";
+import { Search } from "./components/Search";
+
+import "./PosterHall.scss";
+
+export interface PosterHallProps {
+  venue: WithId<GenericVenue>;
+}
+
+export const PosterHall: React.FC<PosterHallProps> = ({ venue }) => {
+  const {
+    posterVenues,
+    titleFilter,
+    setTitleFilter,
+    liveFilter,
+    setLiveFilter,
+  } = usePosters({
+    posterHallId: venue.id,
+  });
+
+  return (
+    <div className="posterhall">
+      <Search
+        setTitleValue={setTitleFilter}
+        titleFilterValue={titleFilter}
+        liveFilterValue={liveFilter}
+        setLiveValue={setLiveFilter}
+      />
```

Comment:
This feels kind of props-drilly?

Reply:
It comes out of `usePosters` and is fed directly to `PosterHallSearch`, where it's used. Given that, I don't think it's prop-drilling as this term refers to a situation when the component doesn't use the props, which it was provided with, other than passing it down the tree.

Reply:
Perhaps an invalid use of the words on my part.. but we are basically passing 2 setters and 2 getters through to the lower component, which feels super hacky, and implies to me that we're likely not handling the 'concerns' in the right places.

Reply:
Not sure if I understand the concern, but what is the way you envision it should be handled?

Reply:
> Not sure if I understand the concern

It's basically tightly coupling that component to the data that need to be passed into it, which feels like the abstractions here may not be ideal.

> but what is the other way you envision it should be handled

I didn't have a specific suggestion in mind here, and to formulate it would require me getting deeper into the code here than I have time/brain space for. Essentially, if we are passing in the state/setters of 2 different things into this component, it feels like that state should probably more belong to that component.

Reply:
Perhaps something with https://react-hook-form.com/ to keep the relevant form state/handlers together or similar?

Reply:
Not sure what you mean by `something`? IMO, the only way to prevent this would be to wrap it into the context

Reply:
> Not sure what you mean by `something`?

React Hook Form's whole purpose seems to be connecting to form elements and managing state and such, in a clean/simple way. I haven't deeply used it to know exactly what 'something', but off the top of my head, perhaps something like `useController`/similar, where a ref is passed into form components and connects/interacts with them in that way:

- https://react-hook-form.com/api/usecontroller

Reply:
[NB] This is NB btw.. it just feels like the same kind of hacky/tech debty/prop drilling types of patterns we're trying to eliminate from the codebase.